src/main/java/seedu/address/logic/commands/AssignCommand.java

```diff
@@ -0,0 +1,170 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Random;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Age;
+import seedu.address.model.person.Detail;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Gender;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.tag.StudyGroupTag;
+
+/**
+ * Assigns the current listed participants into study groups
+ */
+public class AssignCommand extends Command {
+
+    public static final String COMMAND_WORD = "assign";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Assigns persons in the displayed person list randomly to the distinct input study group(s).\n"
+            + "Parameters: "
+            + "STUDY_GROUP_1 [...]\n"
+            + "Example: " + COMMAND_WORD + " P90-Placebo P90-Experimental";
+
+    public static final String MESSAGE_SUCCESS = "Assigned participants successfully!";
+    public static final String MESSAGE_DUPLICATE_STUDYGROUP = "There are duplicate study groups in your input!";
```

Comment:
i vaguely rmb someone said last time to remove ! from our messages. does that still apply? if so, i guess can remove the ! from this message and the below message

Reply:
Not very sure about this, quite a few messages still have punctuation. Do you want to open a bug report to fix it?